app/Repositories/Activity/ActivityRepository.php

```diff
@@ -6,16 +6,190 @@
+                $projects[$searchedModel->id]['playlists'] = [];
+            } else if ($modelName == 'Playlist' && !isset($projects[$searchedModel->project_id]['playlists'][$searchedModel->id])) {
+                $playlistProjectid = $searchedModel->project_id;
+
+                if (!isset($projects[$playlistProjectid])) {
+                    $projects[$playlistProjectid] = $searchedModel->project->attributesToArray();
+                    $projects[$playlistProjectid]['playlists'] = [];
+                }
+
+                $projects[$playlistProjectid]['playlists'][$searchedModel->id] = $searchedModel->attributesToArray();
+                $projects[$playlistProjectid]['playlists'][$searchedModel->id]['activities'] = [];
+            } else if ($modelName == 'Activity') {
+                $activityId = $searchedModel->id;
+                $activityPlaylist = $searchedModel->playlist;
+                $activityPlaylistId = $activityPlaylist->id;
+                $activityPlaylistProjectid = $activityPlaylist->project_id;
+
+                if (!isset($projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId]['activities'][$activityId])) {
+                    if (!isset($projects[$activityPlaylistProjectid])) {
+                        $projects[$activityPlaylistProjectid] = $activityPlaylist->project->attributesToArray();
+                        $projects[$activityPlaylistProjectid]['playlists'] = [];
+                    }
+
+                    if (!isset($projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId])) {
+                        $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId] = $activityPlaylist->attributesToArray();
+                        $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId]['activities'] = [];
+                    }
+
+                    $activityModel = $searchedModel->attributesToArray();
+                    $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId]['activities'][$activityId] = $activityModel;
+                }
+            }
+        }
+
+        return $projects;
+    }
+
+    /**
+     * Get the advance search request
+     *
+     * @param  Request  $request
+     * @return Collection
+     */
+    public function advanceSearchForm($request)
+    {
+        $counts = [];
+        $projectIds = [];
+
+        if ($request->has('userIds') && $request->filled('userIds')) {
+            $userIds = $request->input('userIds');
+
+            $projectIds = Project::whereHas('users', function (Builder $query) use($userIds) {
+                $query->whereIn('id', $userIds);
+            })->pluck('id')->toArray();
+        }
+
+        $searchResultQuery = Activity::searchForm()
+                            ->query($request->input('query', 0))
+                            ->join(Project::class, Playlist::class)
+                            ->aggregate('count_by_index', [
```

Comment:
aggregate?

Reply:
Its function from the Elasticsearch package we are using.

https://github.com/babenkoivan/elastic-scout-driver-plus#aggregate